packages/color/src/Color.ts

```diff
@@ -0,0 +1,331 @@
+ * @memberof PIXI
+ */
+export class Color
+{
+    /**
+     * Default Color object for static uses
+     * @example
+     * import { Color } from 'pixi.js';
+     * Color.default.setValue(0xffffff).toString(); // '#ffffff'
+     */
+    static readonly default = new Color();
+
+    /** Pattern for hex strings */
+    private static readonly HEX_PATTERN = /^(#|0x)?(([a-f0-9]{3}){1,2}([a-f0-9]{2})?)$/i;
+
+    /** Internal color source, from constructor or set value */
+    private _value: Exclude<ColorSource, Color>;
+
+    /** Normalized rgba component, floats from 0-1 */
+    private _components: Float32Array;
+
+    /** Cache color as number */
+    private _int: number;
+
+    /**
+     * @param {PIXI.ColorSource} value - Optional value to use, if not provided, white is used.
+     */
+    constructor(value: ColorSource = 0xffffff)
+    {
+        this._components = new Float32Array(4);
+        this._components.fill(1);
+        this.value = value;
+    }
+
+    /** Read out red value (0 - 1) */
+    get red(): number
+    {
+        return this._components[0];
+    }
+
+    /** Read out green value (0 - 1) */
+    get green(): number
+    {
+        return this._components[1];
+    }
+
+    /** Read out blue value (0 - 1) */
+    get blue(): number
+    {
+        return this._components[2];
+    }
+
+    /** Read out alpha value (0 - 1) */
+    get alpha(): number
+    {
+        return this._components[3];
+    }
+
+    /**
+     * Set the value, suitable for chaining
+     * @param value
+     */
+    setValue(value: ColorSource): this
+    {
+        this.value = value;
+
+        return this;
+    }
+
+    /**
+     * Set the current color source
+     * @type {PIXI.ColorSource}
+     */
+    set value(value: ColorSource)
+    {
+        // Support copying from other Color objects
+        if (value instanceof Color)
+        {
+            this._value = value._value;
+            this._int = value._int;
+            this._components.set(value._components);
+        }
+        else if (this._value !== value)
+        {
+            this.normalize(value);
+            this._value = value;
+        }
+    }
+    get value(): Exclude<ColorSource, Color>
+    {
+        return this._value;
+    }
+
+    /**
+     * Get RGBA color object.
+     * @example
+     * import { Color } from 'pixi.js';
+     * new Color('white').toRgb(); // returns { r: 1, g: 1, b: 1, a: 1 }
+     */
+    toRgba(): RgbaColor
+    {
+        const [r, g, b, a] = this._components;
+
+        return { r, g, b, a };
+    }
+
+    /**
+     * Get RGB color object.
+     * @example
+     * import { Color } from 'pixi.js';
+     * new Color('white').toRgb(); // returns { r: 1, g: 1, b: 1 }
+     */
+    toRgb(): RgbColor
+    {
+        const [r, g, b] = this._components;
+
+        return { r, g, b };
+    }
+
+    /** Get the color as a CSS-style rgba string: `rgba(255,255,255,1.0)`. */
+    toRgbaString(): string
+    {
+        const [r, g, b, a] = this._components;
```

Comment:
I would add a new public `toUint8ClampedArray()` method that returns the `_components` as an array of values beween 0 and 255.

This method could be in turned used here with:

```ts
const [r, g, b] = this.toUint8ClampedArray();
const a = this._components[4]:

return `rgba(${r},${g},${b},${a})`;
```